inc/deployable.class.php

```diff
@@ -0,0 +1,188 @@
+      header("Content-type: $mimeType");
+      header("Last-Modified: $time");
+      header('Accept-Ranges: bytes');
+      header('Content-Length:' . ($end - $begin + 1));
+      header("Content-Range: bytes $begin-$end/$size");
+      header("Content-Transfer-Encoding: binary\n");
+      header('Connection: close');
+
+      $httpStatus = 'HTTP/1.0 200 OK';
+      if ($begin > 0 || $end < $size - 1) {
+         $httpStatus = 'HTTP/1.0 206 Partial Content';
+      }
+      header($httpStatus);
+
+      // Sends bytes until the end of the range or connection closed
+      while (!feof($fileHandle) && $currentPosition < $end && (connection_status() == 0)) {
+         // allow a few seconds to send a few KB.
+         set_time_limit(10);
+         $content = fread($fileHandle, min(1024 * 16, $end - $currentPosition + 1));
+         if ($content === false) {
+            header("HTTP/1.0 500 Internal Server Error", true); // Replace previously sent headers
+            exit(0);
+         }
+         print $content;
+         flush();
+         $currentPosition += 1024 * 16;
+      }
+
+      // Endnow to prevent any unwanted bytes
+      exit(0);
+   }
+
+   /**
+    * @param string $itemtype
+    * @param integer $itemId
+    * @param PluginFlyvemdmTask $task
+    */
+   protected function publishTaskAction($itemtype, $itemId, PluginFlyvemdmTask $task) {
+      $tasks = $task->find("`itemtype`='$itemtype' AND `items_id`='$itemId'");
+      foreach ($tasks as $taskId => $taskRow) {
+         $notifiableType = $taskRow['itemtype_applied'];
+         $notifiable = new $notifiableType();
+         $notifiableId = $taskRow['items_id_applied'];
+         if ($notifiable->getFromDB($notifiableId)) {
+            Toolbox::logInFile('php-errors',
+               "Plugin Flyvemdm : Could not find notifiable id = '$notifiableId'");
+            continue;
+         }
+         if ($task->getFromDB($taskId)) {
+            try {
+               $task->publishPolicy($notifiable);
+               $task->createTaskStatuses($notifiable);
+            } catch (TaskPublishPolicyPolicyNotFoundException $exception) {
+               Session::addMessageAfterRedirect(__($exception->getMessage(), 'flyvemdm'), true,
+                  ERROR);
+            }
+         }
+      }
+   }
+
+   protected function getDeployableURL($deployableItem) {
+      $config = Config::getConfigurationValues('flyvemdm', ['deploy_base_url']);
+      $deployBaseURL = $config['deploy_base_url'];
+      if ($deployBaseURL === null) {
+         return false;
+      }
+      // deployable
+      return $deployBaseURL . '/'.$deployableItem.'/' . $this->fields['filename'];
+   }
+
+   /**
+    * Create a directory
+    * @param string $dir
+    */
+   protected function createEntityDirectory($dir) {
+      if (!is_dir($dir)) {
+         @mkdir($dir, 0770, true);
+      }
+   }
+
+   protected function purgeItem($filename) {
```

Comment:
the name of the method may lead to confusion with purge (meaning deleting am item from database without putting it in the trash). We should find a better name which means we remove a file from the file system.